pandas/core/window.py

```diff
@@ -204,8 +204,21 @@ def __repr__(self):
         )
 
     def __iter__(self):
-        url = "https://github.com/pandas-dev/pandas/issues/11704"
-        raise NotImplementedError("See issue #11704 {url}".format(url=url))
+        self.validate()
+        window = self._get_window()
+        minp = _use_window(self.min_periods, window)
+
+        blocks, obj, index = self._create_blocks()
+        index, indexi = self._get_index(index)
+
+        iterators = []
```

Comment:
Do we need to accumulate the iterators in a list? Can we just `yield`?

Reply:
We could iterate over each iterator and yield each element, though, this might diminish any performance we've gained by implementing our own Iterator.

Would you envision something like this?

```python
for block in blocks:
   for item in iterator_for_block(block):
      yield item
```

We could also lazily create the new iterators

Reply:
I was thinking:

```
for values in blocks:
    for element in libwindow.WindowIterator(values, indexi, window, self.closed, minp):
        yield element
```

Reply:
Yeah exactly :)

I shortened it a bit as I didn't have the source code available.